--- openssl-0.9.7b/e_os.h.orig	Wed Dec  4 09:54:22 2002
+++ openssl-0.9.7b/e_os.h	Tue Aug  5 17:51:15 2003
@@ -79,13 +79,13 @@
 #ifndef DEVRANDOM
 /* set this to a comma-separated list of 'random' device files to try out.
  * My default, we will try to read at least one of these files */
-#define DEVRANDOM "/dev/urandom","/dev/random","/dev/srandom"
+/*#define DEVRANDOM "/dev/urandom","/dev/random","/dev/srandom"*/
 #endif
 #ifndef DEVRANDOM_EGD
 /* set this to a comma-seperated list of 'egd' sockets to try out. These
  * sockets will be tried in the order listed in case accessing the device files
  * listed in DEVRANDOM did not return enough entropy. */
-#define DEVRANDOM_EGD "/var/run/egd-pool","/dev/egd-pool","/etc/egd-pool","/etc/entropy"
+#define DEVRANDOM_EGD "/home/odyssey/mebourne/.egd-pool","/var/run/egd-pool","/dev/egd-pool","/etc/egd-pool","/etc/entropy"
 #endif
 
 #if defined(OPENSSL_SYS_VXWORKS)
