*** semantic-sb.el	Tue Jul 11 15:38:28 2000
--- /homedir/ebourm/config/emacs/extra/semantic-sb.el	Wed Nov 22 10:27:43 2000
***************
*** 103,109 ****
  				     'semantic-sb-token-jump
  				     token t)))
  	  (t
! 	   (if (or (and ttype (or (not (listp ttype)) (car ttype))) edata)
  	       (speedbar-insert-button (if prefix (concat " +" prefix) " +>")
  				       'speedbar-button-face
  				       'speedbar-highlight-face
--- 103,110 ----
  				     'semantic-sb-token-jump
  				     token t)))
  	  (t
! 	   (if edata
! 	       ;(or (and ttype (or (not (listp ttype)) (car ttype))) edata)
  	       (speedbar-insert-button (if prefix (concat " +" prefix) " +>")
  				       'speedbar-button-face
  				       'speedbar-highlight-face
***************
*** 111,130 ****
  				       token t)
  	     (speedbar-insert-button (if prefix (concat "  " prefix) " =>")
  				     nil nil nil nil t))
- 	   (speedbar-insert-button (semantic-token-name token)
- 				   'speedbar-tag-face
- 				   'speedbar-highlight-face
- 				   'semantic-sb-token-jump
- 				   token t)
  	   (cond ((eq type 'variable)
  		  ;; Place array dims here if apropriate.
  		  (if (semantic-token-variable-default token)
  		      (speedbar-insert-button "=" nil nil nil nil t)))
  		 ((eq type 'function)
! 		  (speedbar-insert-button "()" nil nil nil nil t))
  		 ((and (eq type 'include)
  		       (semantic-token-include-system token))
  		  (speedbar-insert-button "<>" nil nil nil nil t))
  		 )))
      ;; This is very bizarre.  When this was just after the insertion
      ;; of the depth: text, the : would get erased, but only for the
--- 112,159 ----
  				       token t)
  	     (speedbar-insert-button (if prefix (concat "  " prefix) " =>")
  				     nil nil nil nil t))
  	   (cond ((eq type 'variable)
+ 		  (speedbar-insert-button (semantic-token-name token)
+ 					  'font-lock-variable-name-face
+ 					  'speedbar-highlight-face
+ 					  'semantic-sb-token-jump
+ 					  token t)
+ 		  (speedbar-insert-button ":" nil nil nil nil t)
+ 		  (let ((type (semantic-token-type token)))
+ 		    (speedbar-insert-button type
+ 					    'font-lock-type-face
+ 					    nil nil nil t))
  		  ;; Place array dims here if apropriate.
  		  (if (semantic-token-variable-default token)
  		      (speedbar-insert-button "=" nil nil nil nil t)))
  		 ((eq type 'function)
! 		  (speedbar-insert-button (semantic-token-name token)
! 					  'font-lock-function-name-face
! 					  'speedbar-highlight-face
! 					  'semantic-sb-token-jump
! 					  token t)
! 		  (speedbar-insert-button (if edata "(...)" "()") nil nil nil nil t)
! 		  (let ((type (semantic-token-type token)))
! 		    (if type
! 			(progn
! 			  (speedbar-insert-button ":" nil nil nil nil t)
! 			  (speedbar-insert-button type
! 						  'font-lock-type-face
! 						  nil nil nil t)))))
  		 ((and (eq type 'include)
  		       (semantic-token-include-system token))
+ 		  (speedbar-insert-button (semantic-token-name token)
+ 					  'font-lock-constant-face
+ 					  'speedbar-highlight-face
+ 					  'semantic-sb-token-jump
+ 					  token t)
  		  (speedbar-insert-button "<>" nil nil nil nil t))
+ 		 (t
+ 		  (speedbar-insert-button (semantic-token-name token)
+ 					  'speedbar-tag-face
+ 					  'speedbar-highlight-face
+ 					  'semantic-sb-token-jump
+ 					  token t))
  		 )))
      ;; This is very bizarre.  When this was just after the insertion
      ;; of the depth: text, the : would get erased, but only for the
***************
*** 185,202 ****
  	     (while parts
  	       (semantic-sb-maybe-token-to-button (car parts) indent)
  	       (setq parts (cdr parts)))))
! 	  ((eq tt 'variable)
! 	   (if type
! 	       (let ((mods (semantic-token-variable-modifiers token)))
! 		 (semantic-sb-maybe-token-to-button type indent "@" mods)))
! 	   ;; default value here
! 	   )
  	  ((eq tt 'function)
! 	   (if type
! 	       (semantic-sb-speedbar-data-line
! 		indent "@"
! 		(if (stringp type) type
! 		  (semantic-token-name type))))
  	   ;; Arguments to the function
  	   (let ((args (semantic-token-function-args token)))
  	     (if (and args (car args))
--- 214,231 ----
  	     (while parts
  	       (semantic-sb-maybe-token-to-button (car parts) indent)
  	       (setq parts (cdr parts)))))
! 	  ;((eq tt 'variable)
! 	  ; (if type
! 	  ;	(let ((mods (semantic-token-variable-modifiers token)))
! 	  ;	  (semantic-sb-maybe-token-to-button type indent "@" mods)))
! 	  ; ;; default value here
! 	  ; )
  	  ((eq tt 'function)
! 	   ;(if type
! 	   ;    (semantic-sb-speedbar-data-line
! 	;	indent "@"
! 	;	(if (stringp type) type
! 	;	  (semantic-token-name type))))
  	   ;; Arguments to the function
  	   (let ((args (semantic-token-function-args token)))
  	     (if (and args (car args))
